android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -143,6 +148,40 @@ class TranslationsMiddleware(
         )
     }
 
+    /**
+     * Removes the site from the list of never translate sites using [scope] and dispatches the result to the
+     * store via [TranslationsAction.SetNeverTranslateSitesAction] or else dispatches the failure
+     * [TranslationsAction.TranslateExceptionAction].
+     *
+     * @param context Context to use to dispatch to the store.
+     * @param tabId Tab ID associated with the request.
+     * @param origin A site origin URI that will have the specified never translate permission set.
+     */
+    private fun removeNeverTranslateSite(
+        context: MiddlewareContext<BrowserState, BrowserAction>,
+        tabId: String,
+        origin: String,
+    ) {
+        engine.setNeverTranslateSpecifiedSite(
+            origin = origin,
+            setting = false,
+            onSuccess = {
+                logger.info("Success requesting never translate sites.")
+            },
+            onError = {
```

Comment:
I'm thinking for `onError`:

```
context.store.dispatch(
    TranslationsAction.OperationRequestedAction(
        tabId = tabId,
        operation = TranslationOperation.FETCH_NEVER_TRANSLATE_SITES,
    ),
)
```

I think that'll ensure the store is in the right state, since the call failed, it is out of sync with the store.

Reply:
oh right, updated to `FETCH_NEVER_TRANSLATE_SITES`